src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -1197,18 +1184,22 @@ struct reduceDB_lt
 void CoreSMTSolver::reduceDB()
 {
     int     i, j;
-    double  extra_lim = cla_inc / learnts.size();    // Remove any clause below this activity
 
     sort(learnts, reduceDB_lt(ca));
     // Don't delete binary or locked clauses. From the rest, delete clauses from the first half
-    // and clauses with activity smaller than 'extra_lim':
+    // and clauses with high glue score
+    int extra = 0;
     for (i = j = 0; i < learnts.size(); i++)
     {
         Clause& c = ca[learnts[i]];
-        if (c.size() > 2 && !locked(c) && (i < learnts.size() / 2 || c.activity() < extra_lim))
+        if (c.getGlue() <= 3) {
+            extra++;
+        }
+        if (c.getGlue() > 3 and c.size() > 2 and not locked(c) and (i+extra < learnts.size() / 2)) {
```

Comment:
From what I understand, glue is always `<= 3`, no? That would make this condition always false (and the one above always true). What is the intention here?

Reply:
That's true.  Previously glue was always `<= 10`, but the test was the same as above.  I seem to have gotten confused with the bits.  I'll commit a fix for this one to see if I get it right now.

Reply:
3 is the cutoff in CMS to "keep forever". Some solvers use 2. It may be interesting to experiment.

Reply:
We could try the cutoff of 2 I guess.  We can then free one bit again for future use, if any :)